lib/tainbox/class_methods.rb

```diff
@@ -26,8 +26,8 @@ def attribute(name, type = nil, **args)
     define_writer = args.fetch(:writer, true)
 
     definer = Tainbox::AttributeDefiner.new(self, name, type, args)
-    definer.define_getter if define_reader
-    definer.define_setter if define_writer
+    definer.define_getter(define_reader)
```

Comment:
Incorrect semantic.

1. Save `define_reader` (at least for back-compatibility)
1. Define `private_reader` or something
1. i think, `attribute :name, reader: :private` serves this case best